src/or/circuitlist.c

```diff
@@ -1844,6 +1845,58 @@ circuit_mark_all_dirty_circs_as_unusable(void)
   SMARTLIST_FOREACH_END(circ);
 }
 
+/**
+ * Report any queued cells on or_circuits as written in our bandwidth
+ * totals, for the specified channel direction.
+ *
+ * When we close a circuit or clear its cell queues, we've read
+ * data and recorded those bytes in our read statistics, but we're
+ * not going to write it. This discrepancy can be used by an adversary
+ * to infer information from our public relay statistics and perform
+ * attacks such as guard discovery.
```

Comment:
Can I ask you to add a line here saying that this function is part of the "fast path" and MUST remain as fast as possible? I'm on a sekrit goal of, over time, identifying clearly all those functions :P.